ts/private/ts_project_worker.js

```diff
@@ -286,6 +281,23 @@ function createProgram(args, initialInputs) {
 
     return { host, program, checkAndApplyArgs };
 
+    function setTimeout(cb) {
+        taskQueue.add(cb);
+        return cb;
+    }
+
+    function clearTimeout(cb) {
+        taskQueue.delete(cb)
+    }
+
+    function applyChanges() {
+        debuglog("Applying changes");
+        for (const task of taskQueue) {
+            taskQueue.delete(task);
+            task();
```

Comment:
If this throws we want to stop the loop? Or are we certain enough it won't throw so it doesn't matter?

Reply:
That's fatal and the worker should die. There is no right way to handle the error that could be raised by the underlying function. Think of this as setTimeout with a callback that throws.

Reply:
It shouldn't throw for user errors. If it throws then either worker does something bad or tsc does.